packages/react-components/react-datepicker/src/components/DatePicker/useDatePicker.tsx

```diff
@@ -554,38 +546,62 @@ export const useDatePicker_unstable = (props: DatePickerProps, ref: React.Ref<HT
         />
       ),
       disabled,
-      label,
       placeholder,
       readOnly: !allowTextInput,
       required: isRequired,
       role: 'combobox',
+      id: textFieldId,
```

Comment:
_nit:_ Keep alphabetical order

```suggestion
      id: textFieldId,
      placeholder,
      readOnly: !allowTextInput,
      required: isRequired,
      role: 'combobox',
```